core/pkg/server/sender.go

```diff
@@ -641,12 +641,15 @@ func (s *Sender) sendUseArtifact(record *service.Record) {
 }
 
 // Inserts W&B-internal information into the run configuration.
-//
-// Uses the given telemetry
 func (s *Sender) updateConfigPrivate() {
-	metrics := []map[string]interface{}(nil)
-	if s.metricSender != nil {
-		metrics = s.metricSender.ConfigMetrics
+	metrics, err := s.runConfigMetrics.ToRunConfigData()
```

Comment:
need to think if we want to update config with partial metric data, as i'm not sure what the UI will do with missing information

Reply:
I don't think the UI uses most of the information anyway. Missing data won't result in incorrect step metric indices.

Reply:
would be nice to verify it with hard coded partial config metrics or just look at the UI code

Reply:
what kinds of problems do you think it could cause? the effect should be as if `define_metric` wasn't used at all

Reply:
if it wasn't at all it's not a problem just that the display would be default instead of using the define_metric hints.
I wonder what happens with partial metric config, if it is just ignored or what? i assume it is safe, just don't know the frontend code implementation so being extra paranoid i guess